{{cookiecutter.project_slug}}/tests.py

```diff
@@ -1,81 +1,187 @@
-                        msg="You didn't add the SKU code in the items list.")
 
-    # Check if redirectToCheckout function call is present
     def test_acceptance_redirect_to_checkout(self):
-        pattern = re.compile(r"stripe.redirectToCheckout", re.I | re.M)
-        res = re.search(pattern, self.dom_str)
-        self.assertTrue(hasattr(res, 'group'), msg="No checkout redirection was found.")
+        """Check if redirectToCheckout function call is present"""
+        pattern = re.compile(
+            r".redirectToCheckout",
+            re.I | re.M
+        )
+        res = re.search(pattern, self.order_html_str)
+
+        self.assertTrue(
+            hasattr(res, 'group'),
+            msg="No checkout redirection was found."
+        )
 
-    # Check if successUrl redirects to order_success.html
     def test_acceptance_success_url(self):
+        """Check if success_url redirects to the /order_success route"""
         pattern = re.compile(
-            r"successUrl: \'(http|https)://(.*)/order_success.html\?session_id=\{CHECKOUT_SESSION_ID\}\'",
-            re.I | re.M)
-        res = re.search(pattern, self.dom_str)
-        self.assertTrue(hasattr(res, 'group'), msg="You didn't define a success URL.")
+            r"success_url=domain_url \+ (\"|')/order_success\?session_id={CHECKOUT_SESSION_ID}(\"|')",
+            re.I | re.M
+        )
+        res = re.search(pattern, self.app_py_str)
+
+        self.assertTrue(
+            hasattr(res, 'group'),
+            msg="You didn't define a success URL."
+        )
 
-    # Check if cancelUrl redirects to order.html
     def test_acceptance_cancel_url(self):
-        pattern = re.compile(r"cancelUrl: \'(http|https)://(.*)/order.html\'",
-                             re.I | re.M)
-        res = re.search(pattern, self.dom_str)
+        """Check if cancel_url redirects to the index route"""
+        pattern = re.compile(
+            r"cancel_url=domain_url \+ (\"|')/(\"|')",
+            re.I | re.M
+        )
+        res = re.search(pattern, self.app_py_str)
+
         self.assertTrue(hasattr(res, 'group'), msg="You didn't define a cancel URL.")
 
 
+class ASTTestCases(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(ASTTestCases, self).__init__(*args, **kwargs)
+        with open('client/order.html', 'r') as file_descriptor:
+            self.order_html_str = file_descriptor.read()
+
+    def test_ast_use_correct_product(self):
+        pattern = re.compile(
+            r"product = (\"|'){{ cookiecutter.product }}(\"|')",
+            re.I | re.M
+        )
+        res = re.search(pattern, self.order_html_str)
+
+        self.assertTrue(hasattr(res, 'group'))
+
+    def test_ast_use_correct_amount(self):
+        pattern = re.compile(
+            r"amount = {{ cookiecutter.amount }}",
+            re.I | re.M
+        )
+        res = re.search(pattern, self.order_html_str)
+
+        self.assertTrue(hasattr(res, 'group'))
+
+
```

Comment:
AST Tests shouldn't live in the same test file as Acceptance/Assessment. It can't count to the final result of the challenge.

Reply:
It doesn't affect the score report though as we only use the tests under Assessment. What's the purpose of having the AST Tests on a separate file?